tst/functions/Mock.Tests.ps1

```diff
@@ -2798,6 +2798,79 @@ Describe 'Mocking using ParameterFilter' {
     }
 }
 
+Describe 'Calls not matching ParameterFilter' {
+    BeforeAll {
+        function demo ($name) {
+            "hello $name"
+        }
+    }
+
+    Context 'When default mock does not exist in parent scope' {
+        It 'Throws by default' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' }
+            { demo -name 'you' } | Should -Throw
+        }
+
+        It 'Calls locally defined default mock' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' }
+            Mock demo { 'default mock' }
+            demo -name 'you' | Should -Be 'default mock'
+        }
+
+        It 'Calls real function when -AllowFallback is used' {
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' } -AllowFallback
+            demo -name 'you' | Should -Be 'hello you'
+        }
+
+        It 'Calls real function when at least one parameterized mock has -AllowFallback' {
+            # TODO: Is this expected? Or should all parameterized mocks allow fallback?
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' } -AllowFallback
+            Mock demo -ParameterFilter { $name -eq 'Wisconsin' } -MockWith { 'mocked2' }
+            demo -name 'you' | Should -Be 'hello you'
+        }
+    }
+
+    Context 'When default mock exists in parent scope' {
+        BeforeAll {
+            Mock demo -MockWith { 'default mock' }
+        }
+        It 'Calls default mock without -AllowFallback' {
+            # TODO: Is this expected? Should -AllowFallback only have effect when there's no default mock at any level?
+            Mock demo -ParameterFilter { $name -eq 'world' } -MockWith { 'mocked' }
+            demo -name 'you' | Should -Be 'default mock'
+        }
```

Comment:
This is likely a common use case where a command returns a default result while some parameterized mocks returns special output. Good as is or any exceptions to this behavior?